src/graph_notebook/authentication/iam_credentials_provider/ec2_metadata_credentials_provider.py

```diff
@@ -20,10 +21,6 @@ def __init__(self):
         self.region = region
 
     def get_iam_credentials(self) -> Credentials:
-        res = requests.get(iam_url)
-        if res.status_code != 200:
-            raise Exception(f'unable to get iam credentials {res.content}')
-
-        js = res.json()
-        creds = Credentials(key=js['AccessKeyId'], secret=js['SecretAccessKey'], token=js['Token'], region=self.region)
-        return creds
+        session = botocore.session.get_session()
```

Comment:
What is going to happen if botocore is not configured on the machine?

Reply:
We should get a NoCredentialsFoundException which would be displayed in the output. 

We could think about adding some kind of validation steps at setup based on the configuration settings. So for `ROLE` as the credentials provider, it would attempt to grab session credentials and fail the setup with follow-up steps before allowing queries to proceed